pkg/pdp/replicator/replicate.go

```diff
@@ -0,0 +1,231 @@
+}
+
+func (r *LocalReplicator) Stop(ctx context.Context) error {
+	return r.queue.Shutdown(ctx)
+}
+
+type storageService struct {
+	claims   claims.Claims
+	blobs    blobs.Blobs
+	pdp      pdp.PDP
+	receipts receiptstore.ReceiptStore
+	id       principal.Signer
+}
+
+func (s *storageService) ID() principal.Signer {
+	return s.id
+}
+
+func (s *storageService) PDP() pdp.PDP {
+	return s.pdp
+}
+
+func (s *storageService) Blobs() blobs.Blobs {
+	return s.blobs
+}
+
+func (s *storageService) Claims() claims.Claims {
+	return s.claims
+}
+
+func (s *storageService) Receipts() receiptstore.ReceiptStore {
+	return s.receipts
+}
+
+func NewSimpleReplicator(c claims.Claims, b blobs.Blobs, p pdp.PDP, r receiptstore.ReceiptStore, i principal.Signer) (*SimpleReplicator, error) {
+	channel := http2.NewHTTPChannel(UploadServiceURL)
+	conn, err := client.NewConnection(UploadServiceDID, channel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to upload service: %w", err)
+	}
+
+	return &SimpleReplicator{
+		uploadServiceConn: conn,
+		storageService: &storageService{
+			claims:   c,
+			blobs:    b,
+			pdp:      p,
+			receipts: r,
+			id:       i,
+		}}, nil
+}
+
+type SimpleReplicator struct {
+	storageService    storage.Service
+	uploadServiceConn client.Connection
+}
+
+func (r *SimpleReplicator) replicate(ctx context.Context, task Task) error {
+	// pull the data from the source
+	replicaResp, err := http.Get(task.Source.String())
+	if err != nil {
+		return fmt.Errorf("http get replication source (%s) failed: %w", task.Source, err)
+	}
+
+	// stream the source to the sink
+	req, err := http.NewRequest(http.MethodPut, task.Sink.String(), replicaResp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to create replication sink request: %w", err)
+	}
+	req.Header = replicaResp.Header
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed http PUT to replicate blob %s from %s to %s failed: %w", task.Blob.Digest, task.Source, task.Sink, err)
+	}
+	// verify status codes
+	if res.StatusCode >= 300 || res.StatusCode < 200 {
+		err := fmt.Errorf("unsuccessful http PUT to replicate blob %s from %s to %s status code %d", task.Blob.Digest, task.Source, task.Sink, res.StatusCode)
+		resData, err := io.ReadAll(res.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read replication sink response body: %w", err)
+		}
+		return fmt.Errorf("response body: %s: %w", resData, err)
+	}
+
+	// TODO this is a really gross way to have a dep, but can refactor later
+	acceptResp, err := storage.BlobAccept(ctx, r.storageService, &storage.BlobAcceptRequest{
+		Space: task.Space,
+		Blob:  task.Blob,
+		Put: blob.Promise{
+			UcanAwait: blob.Await{
+				Selector: ".out.ok",
+				// TODO IDK what this does, or what to put here, or if we even need it?
+				Link: task.Invocation.Link(),
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to accept replication source blob %s: %w", task.Blob.Digest, err)
+	}
```

Comment:
Taking a dependency on the `BlobAccept` method from the ucan package isn't going to work here. I am attempting to avoid re-writing the same code, so may move some of these methods to a different package that can be imported by both the ucan server, and things like the replicator as I image this won't be the only time we need to re-use this logic.